### ISSUE TITLE
Image artifacts

### ISSUE THREAD

**Issue by Darrenxc** [Lotayou/Face-Renovation#12]
Hi, 
I followed your instructions to train the network, However, there are some artifacts in generated images 
These artifacts normally happen on where relatively more white pixel locate at, such as eyes, tip of nose.
there are some pics for details  
![epoch023_iter227400_input_label](https://user-images.githubusercontent.com/57433102/88782562-4f256c00-d1c0-11ea-825e-849b38f3cfc8.png)
![epoch023_iter227400_real_image](https://user-images.githubusercontent.com/57433102/88782566-50ef2f80-d1c0-11ea-9f68-c90286c5650e.png)
![epoch023_iter227400_synthesized_image](https://user-images.githubusercontent.com/57433102/88782567-52205c80-d1c0-11ea-9a71-fde7b168f200.png)

![epoch023_iter228800_synthesized_image](https://user-images.githubusercontent.com/57433102/88783624-995b1d00-d1c1-11ea-921e-36495fa8fe0d.png)

Can I ask the reason why these artifacts happened, and how can I solve these.
Thank you!

**Comment by Lotayou** [comment]
I could use more information, how did you process the training data, with what type of degradation? Is the testing data prepared the same way as the training data?

From what I gather, the issue is most likely inherited from the original SPADE network. In fact, we observed even more severe artifacts when using the vanilla SPADE with face parsing maps as the input (Never mind the 16x mosaic on the left column, I just haven't got time to change the visualization script yet):
![image](https://user-images.githubusercontent.com/33449901/89620914-fee69200-d8c2-11ea-8a90-b158bd4bf1c1.png)
![image](https://user-images.githubusercontent.com/33449901/89620899-f55d2a00-d8c2-11ea-9f70-d4ef5f931582.png)

and here is the renovation result using our network, where the input is the left 16x mosaic image, you can still notice the unusual white dot on the woman's nose tip, but for most cases the result is acceptable:
![image](https://user-images.githubusercontent.com/33449901/89621267-a06de380-d8c3-11ea-88be-6357a4c59bce.png)
![image](https://user-images.githubusercontent.com/33449901/89621092-55ec6700-d8c3-11ea-9fdc-b571e78dc976.png)

As to the exact cause of such artifact, I think it could be similar as the ones reported in [StyleGAN2](https://arxiv.org/abs/1912.04958), and maybe applying the corresponding fix could help (for instance, remove the bias normalization). Let me know if you have any progress:)

**Comment by Darrenxc** [comment]
following the stylegan2 instructions about removing instance norm can eliminate this artifact,  I also use mobilev2,v3 to compress this model and face parsing as an additional input, but I notice that you have mentioned a bug in your code that the predicted images are not shown as expected when we set the model to eval model. Do you find a way to solve this problem? I try to fine-tune the entire model for a large of epochs with a very low learning rate, However, it still not good as I expected.

**Comment by Lotayou** [comment]
```
but I notice that you have mentioned a bug in your code that the predicted images are not shown as expected when we set the model to eval model.
```

This is a Pytorch [bug](https://discuss.pytorch.org/t/spectral-norm-in-eval-mode/26223) where the spectral normalization layer in SPADE is not working under eval mode in dataparallel. A possible fix is suggested at [here](https://github.com/pytorch/pytorch/pull/12671)
